ui/round/src/ctrl.ts

```diff
@@ -169,12 +169,19 @@ export default class RoundController implements MoveRootCtrl {
   };
 
   private onMove = (orig: Key, dest: Key, captured?: Piece) => {
-    if (captured || this.enpassant(orig, dest)) {
-      if (this.data.game.variant.key === 'atomic') {
-        site.sound.play('explosion');
-        atomic.capture(this, dest);
-      } else site.sound.move({ name: 'capture', filter: 'game' });
-    } else site.sound.move({ name: 'move', filter: 'game' });
+    if ((captured || this.enpassant(orig, dest)) && this.data.game.variant.key === 'atomic') {
+      site.sound.play('explosion');
+      atomic.capture(this, dest);
+      return;
+    }
+    const fen =
+      this.ply === 0
+        ? (this.data.game.initialFen ?? 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR')
+        : this.stepAt(this.ply - 1).fen;
+    const san = sanOf(readFen(fen), orig + dest);
```

Comment:
Pretty sure `sanOf(readFen(fen), move)` is expensive to run at such a critical moment; that's why the server sends the SAN to the client.
Also I'm not sure that works with variants.

If we want to play a sound before the server sends the SAN back, it should be possible to do so with computing the SAN.

Reply:
sanOf(readFen) takes around 2-3 microseconds.

pentatonic uses information such as castles and check which are not readily available from chessground. is it worth separate code paths to shave off a computation we could do a thousand times in a loop without dropping a frame?

Reply:
nope.